samza-core/src/test/java/org/apache/samza/processor/TestStreamProcessor.java

```diff
@@ -43,53 +48,85 @@
 import static org.mockito.Mockito.when;
 
 public class TestStreamProcessor {
+  private Map<String, Boolean> processorListenerState;
```

Comment:
A ConcurrentMap would be appropriate here if there is a chance that we are (or will) invoke a state transition from another thread.